components/brave_ads/browser/BUILD.gn

```diff
@@ -54,13 +54,32 @@ source_set("browser") {
       "notification_helper.h",
     ]
 
+    deps += [
```

Comment:
why were all these other blocks moved? It confuses the change and seems likely to have more merge conflicts

Reply:
I rearranged them because we had separate per-platform `if` statements for `sources` and `deps`. I changed it to have per-platform `sources` and `deps` together under one `if` statement for each platform. The only actual change here is the addition of `//ui/gfx/x` as a `dep` on Linux.